### PR TITLE
Setup the project foundation, MVVM architecture, implemented biometric and Oauth authentication

### PR DESCRIPTION
**Pull request by 1525cb1c708d145ecc646db05229b9_bcgovent**
The project is shaping up really well.
There are some areas where we need to work on like Constants, Localization, Environments, Test Cases, Design, Pipelines, Code scanning etc.
It is going to be pretty hard to accomplish these given the March 31st deadline. I'll try to concentrate on these areas in the future PRs.
Here is a recording to help understand the end product of this PR well. TIA!

https://github.com/user-attachments/assets/c55fa414-f540-4c7a-9a30-6d967e48b25d

### CODE REVIEW

File: hestia/hestia/ViewModels/OpenIdAuthenticationViewModel.cs [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+		{
+			this.authClient = authClient;
+		}
+
+        public async Task<Result> Authenticate()
+        {
+            var loginResult = await authClient.LoginAsync();
+            if (loginResult.IsError)
+            {
+                //Console.WriteLine($"loginResult.ErrorDescription: {loginResult.ErrorDescription}");
```
Comment (tschbc) [inline hestia/hestia/ViewModels/OpenIdAuthenticationViewModel.cs:23]: I forgot to write about this in our coding conventions: please don't commit commented code—if it was for debugging or for "historical" purposes.

File: hestia/hestia/ViewModels/OpenIdAuthenticationViewModel.cs [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+                return new Result(isError: true, error: loginResult.Error, errorDescription: loginResult.ErrorDescription);
+            }
+            else
+            {
+                //Console.WriteLine("loginResult");
```
Comment (tschbc) [inline hestia/hestia/ViewModels/OpenIdAuthenticationViewModel.cs:28]: Same deal as above: please remove commented code.

File: hestia/hestia/ViewModels/OpenIdAuthenticationViewModel.cs [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+                //Console.WriteLine($"loginResult: AccessToken: {loginResult.AccessToken}");
+                //Console.WriteLine($"loginResult: RefreshToken: {loginResult.RefreshToken}");
+                return new Result(isError: false, error: null, errorDescription: null);
+            }
```
Comment (tschbc) [inline hestia/hestia/ViewModels/OpenIdAuthenticationViewModel.cs:32]: This if block can be reduced:

```suggestion
            return new Result(loginResult.isError, loginResult.Error, loginResult.ErrorDescription);
```

File: hestia/hestia/Views/LandingPage.xaml.cs [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+        BindingContext = viewModel;
+        this.viewModel = viewModel;
+        this.router = router;
+        ListenToViewModelProperties();
+	}
+
+    protected override void OnLoadAsync()
+    {
+        base.OnLoadAsync();
+        //TriggerRouteUpdate();
```
Comment (tschbc) [inline hestia/hestia/Views/LandingPage.xaml.cs]: Remove commented code

File: hestia/hestia/Views/LandingPage.xaml.cs [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+        router.RouteTo(solvedRoute);
+    }
+
+    void ListenToViewModelProperties()
+    {
+        viewModel.PropertyChanged += (sender, args) =>
+        {
+            if (args.PropertyName.Equals(nameof(viewModel.BackFromPage)))
+            {
+                //Shell.Current.CurrentState
```
Comment (tschbc) [inline hestia/hestia/Views/LandingPage.xaml.cs]: Remove commented code

**Comment by tschbc** [comment]
I noticed that indentation is broken in a lot of the files. You can [run a tool in Visual Studio](https://stackoverflow.com/questions/5755942/how-do-you-auto-format-code-in-visual-studio) to automatically fix that.

File: hestia/hestia/Routers/DeviceAuthenticationRouter.cs [hunk 6]
```diff
@@ -0,0 +1,40 @@
...
+            }
+            switch (result)
+            {
+                case DeviceAuthenticator.Result.NotConfigured:
+                    _ = Shell.Current.DisplayAlert("Unprotected Device!",
+                        "Please secure your device by setting up device lock and try again.", "Ok");
+                    break;
+                case DeviceAuthenticator.Result.Successful:
+                    ((AppShell)Shell.Current).
+                        GoToAsyncRequest($"..?navigatingBackFromPage={nameof(DeviceAuthenticationPage)}");
```
Comment (tschbc) [inline hestia/hestia/Routers/DeviceAuthenticationRouter.cs]: This is a perfect place to refactor this statement into its own private function.

**Comment by sAYfj14cKfzD6ZhKDYtT0aQx8Z6DemzBoyR9jJ2** [comment]
@Todd-Scharien_bcgovent I just formatted the code.

**Review by tschbc — APPROVED** [review]

**Pull request merged**